src/batch-submitter/state-batch-submitter.ts

```diff
@@ -170,7 +170,20 @@ export class StateBatchSubmitter extends BatchSubmitter {
     startBlock: number,
     endBlock: number
   ): Promise<Bytes32[]> {
-    const batch: Bytes32[] = []
+    const blockRange = endBlock - startBlock
+    const batch: Bytes32[] = await bPromise.map(
+      [...Array(blockRange).keys()],
+      async (i: number) => {
+        const block = (await this.l2Provider.getBlockWithTransactions(
+          startBlock + i
+        )) as L2Block
+        if (block.transactions[0].from === this.fraudSubmissionAddress) {
+          this.fraudSubmissionAddress = 'no fraud'
+          return '0xbad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1'
+        }
+        return block.stateRoot
+      }
+    )
```

Comment:
Shouldn't the `for` loop below also be removed?

Reply:
going to clean those up as i add tests